miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditProjectController.java

```diff
@@ -259,18 +267,25 @@ public ModelAndView byProjectShortName(@PathVariable String shortName, ModelMap
   @RequestMapping(value = "/{projectId}", method = RequestMethod.GET)
   public ModelAndView setupForm(@PathVariable Long projectId, ModelMap model) throws IOException {
     try {
+      List<Issue> issues = Collections.emptyList();
       User user = securityManager.getUserByLoginName(SecurityContextHolder.getContext().getAuthentication().getName());
       Project project = null;
       if (projectId == ProjectImpl.UNSAVED_ID) {
         project = new ProjectImpl(user);
         model.put("title", "New Project");
       } else {
         project = projectService.getProjectById(projectId);
+        if (project == null) {
+          throw new NotFoundException("No project found for ID " + projectId.toString());
+        }
         model.put("title", "Project " + projectId);
+        try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)